storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbIterator.java

```diff
@@ -93,6 +100,9 @@ public boolean hasNext() {
   }
 
   private void assertOpen() {
+    if (this.isDatabaseClosed.get()) {
+      throw new ShuttingDownException();
+    }
     if (closed.get()) {
       throw new IllegalStateException("Attempt to update a closed transaction");
```

Comment:
(optional) While we're here:
```suggestion
      throw new IllegalStateException("Attempt to update a closed " + this.getClass().getSimpleName());
```